octopusdeploy_framework/resource_team_mappers.go

```diff
@@ -0,0 +1,265 @@
+	model.ID = types.StringValue(team.GetID())
+	model.Name = types.StringValue(team.Name)
+	model.Description = types.StringValue(team.Description)
+	model.SpaceID = types.StringValue(team.SpaceID)
+
+	if len(team.MemberUserIDs) > 0 {
+		model.Users = flattenStringSet(team.MemberUserIDs, model.Users)
+	} else {
+		model.Users = types.SetNull(types.StringType)
+	}
+
+	if len(team.ExternalSecurityGroups) > 0 {
+		model.ExternalSecurityGroup = mapExternalSecurityGroupsResourceToState(ctx, team.ExternalSecurityGroups)
+	} else {
+		objectType := types.ObjectType{
+			AttrTypes: map[string]attr.Type{
+				"id":                  types.StringType,
+				"display_name":        types.StringType,
+				"display_id_and_name": types.BoolType,
+			},
+		}
+		model.ExternalSecurityGroup = types.ListNull(objectType)
+	}
+
+	// Temporary workaround to avoid errors due to differences in validation behaviour between SDKv2 and TPF providers.
+	// TODO: mark as read-only after deprecation phase.
+	if model.CanBeDeleted.IsUnknown() {
+		model.CanBeDeleted = types.BoolNull()
+	} else {
+		model.CanBeDeleted = model.CanBeDeleted
+	}
+	if model.CanBeRenamed.IsUnknown() {
+		model.CanBeRenamed = types.BoolNull()
+	} else {
+		model.CanBeRenamed = model.CanBeRenamed
+	}
+	if model.CanChangeMembers.IsUnknown() {
+		model.CanChangeMembers = types.BoolNull()
+	} else {
+		model.CanChangeMembers = model.CanChangeMembers
+	}
+	if model.CanChangeRoles.IsUnknown() {
+		model.CanChangeRoles = types.BoolNull()
+	} else {
+		model.CanChangeRoles = model.CanChangeRoles
+	}
+
+	userRoles, err := client.Teams.GetScopedUserRoles(*team, core.SkipTakeQuery{})
+	if err == nil {
+		model.UserRole = mapUserRoleSetResourceToState(ctx, userRoles.Items, model.UserRole)
+	}
+
+	return model
+}
+
+func mapExternalSecurityGroupsStateToResource(ctx context.Context, securityGroups types.List) []core.NamedReferenceItem {
+	if securityGroups.IsNull() || securityGroups.IsUnknown() {
+		return nil
+	}
+
+	securityGroupElements := securityGroups.Elements()
+	externalGroups := make([]core.NamedReferenceItem, len(securityGroupElements))
+
+	for i, securityGroupElement := range securityGroupElements {
+		securityGroupAttributes := securityGroupElement.(types.Object).Attributes()
+
+		externalGroups[i] = core.NamedReferenceItem{
+			ID:               securityGroupAttributes["id"].(types.String).ValueString(),
+			DisplayName:      securityGroupAttributes["display_name"].(types.String).ValueString(),
+			DisplayIDAndName: securityGroupAttributes["display_id_and_name"].(types.Bool).ValueBool(),
+		}
+	}
+
+	return externalGroups
+}
+
+func mapExternalSecurityGroupsResourceToState(ctx context.Context, groups []core.NamedReferenceItem) types.List {
+	if len(groups) == 0 {
+		objectType := types.ObjectType{
+			AttrTypes: map[string]attr.Type{
+				"id":                  types.StringType,
+				"display_name":        types.StringType,
+				"display_id_and_name": types.BoolType,
+			},
```

Comment:
Can this be centralised somewhere please, right now it's duplicated three times in this function